src/Run/Except/Checked.purs

```diff
@@ -0,0 +1,51 @@
+-- | ## Extensible checked exceptions with polymorphic variants
+-- |
+-- | This module provides helpers for using `Variant` with `Except`. When
+-- | combined, we get a mechanism for extensible, checked exceptions. That
+-- | is, exceptions can be defined and used anywhere, and handled as needed.
+-- | Handling an exception eliminates it from the type, giving us proof
+-- | that it no longer occurs.
+module Run.Except.Checked
+  ( EXCEPTV
+  , handleError
+  , safe
+  ) where
+
+import Prelude
+
+import Data.Either (either)
+import Data.Variant (class VariantMatchCases, Variant, case_, onMatch)
+import Run (Run, expand)
+import Run.Except (EXCEPT, runExcept, throw)
+import Type.Row (class RowToList)
+
+type EXCEPTV exc = EXCEPT (Variant exc)
+
+-- | Catches and eliminates exceptions given a record of handlers.
+-- | Unhandled exceptions are re-propragated. Record fields map
+-- | to the label for the exception being handled.
+-- |
+-- | An example for handling HTTP exceptions might be:
+-- | ```purescript
+-- | request # handleError
+-- |   { httpNotFound: \_ -> ...
+-- |   , httpServerError: \error -> ...
+-- |   }
+-- | ```
+handleError
+  ∷ ∀ m handlers excHandled excIn excOut rl r a
+  . RowToList handlers rl
+  ⇒ VariantMatchCases rl excHandled (Run (except :: EXCEPTV excOut | r) a)
+  ⇒ Union excHandled excOut excIn
+  ⇒ Union r m (except :: EXCEPTV excOut | r)
```

Comment:
I'm unsure about this bit of the type.

The `expand` below forced this constraint. Is it right? Maybe there's another way to do the same thing?

Reply:
Is the expand necessary if you account for duplicate labels? That is, something like
```purescript
  ...
  → Run (except :: EXCEPTV excIn, except :: EXCEPTV excOut | r) a
  → Run (except :: EXCEPTV excOut | r) a
```

Reply:
Looks like dropping the `expand` and changing to the signature you suggest allows dropping the `Union r m (except :: EXCEPTV excOut | r)` constraint.

Which would you prefer: with expand and constraint, or without?